Gems/Blast/Code/Source/Family/BlastFamilyImpl.cpp

```diff
@@ -102,95 +95,87 @@ namespace Blast
     {
+        AZ_Assert(m_tkFamily, "No TkFamily created for this BlastFamily.");
+        if (!m_isSpawned)
+        {
+            return;
+        }
+
         // Intentional copy here as we will use this set to delete actors from ActorTracker itself
         AZStd::unordered_set<BlastActor*> toDelete = m_actorTracker.GetActors();
         DestroyActors(toDelete);
 
-        if (m_tkFamily)
-        {
-            m_tkFamily->removeListener(*this);
-        }
+        m_tkFamily->removeListener(*this);
         m_isSpawned = false;
     }
 
     void BlastFamilyImpl::HandleEvents(const Nv::Blast::TkEvent* events, uint32_t eventCount)
     {
         AZ_PROFILE_FUNCTION(Physics);
 
-        AZStd::vector<BlastActorDesc> newActors;
-        AZStd::unordered_set<BlastActor*> actorsToDelete;
-
         for (uint32_t i = 0; i < eventCount; ++i)
         {
             const Nv::Blast::TkEvent& event = events[i];
             switch (event.type)
             {
             case Nv::Blast::TkEvent::Split:
                 {
-                    HandleSplitEvent(event.getPayload<Nv::Blast::TkSplitEvent>(), newActors, actorsToDelete);
+                    AZStd::vector<BlastActorDesc> newActorsDesc;
+                    AZStd::unordered_set<BlastActor*> actorsToDelete;
+
+                    HandleSplitEvent(event.getPayload<Nv::Blast::TkSplitEvent>(), newActorsDesc, actorsToDelete);
+
+                    DestroyActors(actorsToDelete);
+                    CreateActors(AZStd::move(newActorsDesc));
                     break;
                 }
             default:
                 break;
             }
         }
-
-        DestroyActors(actorsToDelete);
-        CreateActors(AZStd::move(newActors));
     }
 
     void BlastFamilyImpl::HandleSplitEvent(
-        const Nv::Blast::TkSplitEvent* splitEvent, AZStd::vector<BlastActorDesc>& newActors,
+        const Nv::Blast::TkSplitEvent* splitEvent, AZStd::vector<BlastActorDesc>& newActorsDesc,
         AZStd::unordered_set<BlastActor*>& actorsToDelete)
     {
         AZ_PROFILE_FUNCTION(Physics);
 
-        AZ_Assert(splitEvent, "Received null TkSplitEvent from the Blast library.");
         if (!splitEvent)
         {
+            AZ_Error("Blast", false, "Received null TkSplitEvent from the Blast library.");
             return;
         }
 
-        const uint32_t newActorsCount = splitEvent->numChildren;
-
-        BlastActor* parentActor = nullptr;
-        AzPhysics::SimulatedBody* parentBody = nullptr;
-        AZ_Assert(splitEvent->parentData.userData, "Parent actor in split event must have user data.");
         if (!splitEvent->parentData.userData)
         {
+            AZ_Error("Blast", false, "Parent actor in split event must have user data.");
             return;
         }
 
-        parentActor = reinterpret_cast<BlastActor*>(splitEvent->parentData.userData);
+        BlastActor* parentActor = reinterpret_cast<BlastActor*>(splitEvent->parentData.userData);
```

Comment:
question: Can this in fact just be a `static_cast`? If this is a `void*` I believe the `static_cast` should suffice (could be wrong but worth checking if you have a moment)

Reply:
I'll give it a go.

Reply:
It worked, changed.

Reply:
Nice thanks!